pkg/controller/atlasproject/alert_configurations_test.go

```diff
@@ -0,0 +1,406 @@
+package atlasproject
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/toptr"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.uber.org/zap/zaptest"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/customresource"
+)
+
+type alertConfigurationClient struct {
+	ListFunc func(projectID string) ([]mongodbatlas.AlertConfiguration, *mongodbatlas.Response, error)
+}
+
+func (c *alertConfigurationClient) Create(_ context.Context, _ string, _ *mongodbatlas.AlertConfiguration) (*mongodbatlas.AlertConfiguration, *mongodbatlas.Response, error) {
+	return nil, nil, nil
+}
+
+func (c *alertConfigurationClient) EnableAnAlertConfig(_ context.Context, _ string, _ string, _ *bool) (*mongodbatlas.AlertConfiguration, *mongodbatlas.Response, error) {
+	return nil, nil, nil
+}
+
+func (c *alertConfigurationClient) GetAnAlertConfig(_ context.Context, _ string, _ string) (*mongodbatlas.AlertConfiguration, *mongodbatlas.Response, error) {
+	return nil, nil, nil
+}
+
+func (c *alertConfigurationClient) GetOpenAlertsConfig(_ context.Context, _ string, _ string) ([]mongodbatlas.AlertConfiguration, *mongodbatlas.Response, error) {
+	return nil, nil, nil
+}
+
+func (c *alertConfigurationClient) List(_ context.Context, projectID string, _ *mongodbatlas.ListOptions) ([]mongodbatlas.AlertConfiguration, *mongodbatlas.Response, error) {
+	return c.ListFunc(projectID)
+}
+
+func (c *alertConfigurationClient) ListMatcherFields(_ context.Context) ([]string, *mongodbatlas.Response, error) {
+	return nil, nil, nil
+}
+
+func (c *alertConfigurationClient) Update(_ context.Context, _ string, _ string, _ *mongodbatlas.AlertConfiguration) (*mongodbatlas.AlertConfiguration, *mongodbatlas.Response, error) {
+	return nil, nil, nil
+}
+
+func (c *alertConfigurationClient) Delete(_ context.Context, _ string, _ string) (*mongodbatlas.Response, error) {
+	return nil, nil
+}
```

Comment:
I think this mock code must go to its own package, or at least a separate file.

Note we will use these mocks from many tests. Not sure if we want mocks on all packages or a package with all mocks.

Reply:
Later, they will all be extracted to (probably) their own package

Reply:
I would already put them on their own file, even if we keep them in the same package. When we move them later it will be clearer what is being done.

Reply:
This task is quite big and has been running for a long time. I have already created a ticket to follow up on the mocks.